upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go

```diff
@@ -138,18 +126,40 @@ func (t *LaunchTemplate) RenderAWS(c *awsup.AWSAPITarget, a, ep, changes *Launch
 			InstanceInterruptionBehavior: t.InstanceInterruptionBehavior,
 			MaxPrice:                     fi.String(t.SpotPrice),
 		}
-		lc.InstanceMarketOptions = &ec2.LaunchTemplateInstanceMarketOptionsRequest{
+		data.InstanceMarketOptions = &ec2.LaunchTemplateInstanceMarketOptionsRequest{
 			MarketType:  fi.String("spot"),
 			SpotOptions: s,
 		}
 	}
 	// @step: attempt to create the launch template
-	if _, err = c.Cloud.EC2().CreateLaunchTemplate(input); err != nil {
-		return fmt.Errorf("error creating LaunchTemplate: %v", err)
+	if a == nil {
+		input := &ec2.CreateLaunchTemplateInput{
+			LaunchTemplateName: t.Name,
+			LaunchTemplateData: data,
+			TagSpecifications: []*ec2.TagSpecification{
+				{
+					ResourceType: aws.String(ec2.ResourceTypeLaunchTemplate),
+					Tags:         tags,
+				},
+			},
+		}
+		output, err := c.Cloud.EC2().CreateLaunchTemplate(input)
+		if err != nil || output.LaunchTemplate == nil {
+			return fmt.Errorf("error creating LaunchTemplate %q: %v", fi.StringValue(t.Name), err)
+		}
+		e.ID = output.LaunchTemplate.LaunchTemplateId
+	} else {
+		// TODO: Update the LaunchTemplate tags
```

Comment:
I think implementing this might block merging.

Reply:
Hmm, already implemented it in https://github.com/kubernetes/kops/pull/10151/commits/852c0e2a8f4c376909d413db500fa798807239f2 and seems to work without issues. Nothing actually reads these tags after creation, they are just used for deleting the LT when needed.